core/modules/qana/QservRestrictorPlugin.cc

```diff
@@ -64,27 +64,354 @@
+            _generator = std::make_shared<AreaGenerator>("s2PtInEllipse",
+                                                         5,
+                                                         r._params
+                                                         );
+        } else if (r._name == "qserv_areaspec_poly") {
+            const int use_string = AreaGenerator::USE_STRING;
+            _generator = std::make_shared<AreaGenerator>("s2PtInCPoly",
+                                                         use_string,
+                                                         r._params
+                                                         );
+        } else {
+            throw qana::AnalysisBug("Unmatched restriction spec: " + _name);
+        }
+    }
+    std::string _name;
+    std::vector<double> _params;
+    std::shared_ptr<Generator> _generator;
+};
+
+
+query::BoolTerm::Ptr makeCondition(std::shared_ptr<const query::QsRestrictor> const restr,
+                                   RestrictorEntry const& restrictorEntry) {
+    Restriction r(*restr);
+    return r.generate(restrictorEntry);
+}
+
+
+/**
+ * @brief Determine if the given ValueExpr represents a func that is one of the scisql functions that
+ *        starts with `scisql_s2PtIn` that represents an area restrictor.
+ *
+ * This is a helper function for handleScisqlRestrictors.
+ *
+ * @param valueExpr The ValueExpr to check.
+ * @return true if the ValueExpr is a function and starts with `scisql_s2PtIn` else false.
+ */
+bool isScisqlAreaFunc(query::ValueExpr const& valueExpr) {
+    if (not valueExpr.isFunction()) return false;
+    auto&& funcExpr = valueExpr.getFunction();
+    if (funcExpr->getName().find("scisql_s2Pt") != 0) return false;
+    return true;
+}
+
+
+/**
+ * @brief If there is exactly one scisql area restrictor in the top level AND of the where clause,
+ *        return it.
+ *
+ * @param whereClause The WHERE clause to look in.
+ * @return std::shared_ptr<const query::FuncExpr> The scisql area restrictor functio FuncExpr if there
+ *         was exactly one, else nullptr.
+ */
+std::shared_ptr<const query::FuncExpr> extractSingleScisqlAreaFunc(query::WhereClause const& whereClause) {
+    auto topLevelAnd = whereClause.getRootAndTerm();
+    std::shared_ptr<const query::FuncExpr> scisqlFunc;
+    if (nullptr == topLevelAnd) return nullptr;
+    for (auto const& boolTerm : topLevelAnd->_terms) {
+        auto boolFactor = std::dynamic_pointer_cast<const query::BoolFactor>(boolTerm);
+        if (nullptr == boolFactor) continue;
+        for (auto boolFactorTerm : boolFactor->_terms) {
+            auto compPredicate = std::dynamic_pointer_cast<const query::CompPredicate>(boolFactorTerm);
+            if (nullptr == compPredicate) continue;
+            if (compPredicate->op != query::CompPredicate::EQUALS_OP) continue;
+            for (auto const& valueExpr: {compPredicate->left, compPredicate->right})
```

Comment:
Please, put `{` in the end of the line and indent subsequent `if`. Otherwise the code looks confusing (the first time one reads it 😸 ), and the subsequent if looks like just an ordinary standalone statement.

Reply:
oh man that was a typo not having braces for the for. I'm just lucky I got away with it because there was exctly one 'if' statement in the loop 🤣